core/arch/riscv64/memfuncs.asm

```diff
@@ -44,30 +44,130 @@ START_FILE
  */
         DECLARE_FUNC(memcpy)
 GLOBAL_LABEL(memcpy:)
-/* TODO i#3544: Naive version, optimize it. */
-        mv       t1, ARG1
-        beqz     ARG3, 2f
-1:      lbu      t2, 0(ARG2)
-        addi     ARG2, ARG2, 1
-        sb       t2, 0(t1)
-        addi     t1, t1, 1
-        addi     ARG3, ARG3, -1
-        bnez     ARG3, 1b
-2:      ret
+        addi     a5, x0, 32
+        mv       t0, ARG2 /* save for return */
+copy32_:
+        blt      ARG3, a5, copy0_32
```

Comment:
(OK, this loop copies 32 bytes at a time; we have sequences for smaller sizes.) (Using parens to mean no reply or action requested.)